source/tutorials/zfs.rst

```diff
@@ -0,0 +1,299 @@
+   sudo dkms install -m zfs -v 0.8.4
+
+This will install the zfs kernel modules to:
+
+    /usr/lib/modules/<kernel-name>/extra/zfs
+
+This will install the zfs userspace tools to:
+::
+
+   + /usr/local/
+   |--+ bin/
+      |--zvol_wait
+      |--zgenhostid
+      |--raidz_test
+   |--+ etc/
+      |--+ zfs/
+         |--* zed.d/
+         |--+ zpool.d/
+         |--zfs-functions
+   |--+ include/libzfs/ [contents omitted]
+   |--+ lib/
+   |--+ libexec/
+      |--+ zfs/
+         |-- zpool.d/
+         |-- zed.d
+   |--+ share/zfs/ [contents omitted]
+   |--+ sbin/
+      |--fsck.zfs
+      |--zpool
+      |--zdb
+      |--zed
+      |--zfs
+      |--zhack
+      |--zinject
+      |--zpool
+      |--ztest
+      |--zstreamdump
+   |--+ src/
+      |--+ zfs-0.8.4/
+      |--+ spl-0.8.4/
+
+
+Load the new kernel module:
+
+.. code-block: bash
+
+   sudo modprobe zfs
+
+
+Set up systemd
+==============
+
+You should now have these unit files available:
+
+```
+zfs-env-bootfs.service
+zfs-zed.service
+zfs-import-cache.service
+zfs-import-scan.service
+zfs-mount.service
+zfs-share.service
+zfs-volume-wait.service
+```
+
+If you want to use ZFS automatic zpool import and filesystem
+mount services, copy the systemd.unit files into /etc and enable them:
+
+.. code-block:: bash
+
+   sudo cp ./etc/systemd/system/*.service /etc/systemd/system/
+   sudo cp ./contrib/dracut/90zfs/zfs-env-bootfs.service /etc/systemd/system/
+
+   systemctl enable zfs-import-cache
+   systemctl enable zfs-import.target
+
+   systemctl enable zfs-mount
+   systemctl enable zfs.target
+
+If you prefer to use the zfs-mount-generator instead of zfs-mount, refer to the
+Arch Linux Guide's `ZFS Mount Generator`_ section for details. 
+
+Load the kernel module at boot
+==============================
+
+The ZFS module will not load automatically at boot. Load the zfs.ko module
+at boot time by with systemd.
+
+First, allow unsigned modules:
+
+    echo "module.sig_unenforce" | sudo tee /etc/kernel/cmdline.d/allow-unsigned-modules.conf
+
+Systemd uses the `/etc/modules-load.d/` directory to load out-of-tree kernel modules.
+Make sure that the directory exists:
+
+    sudo mkdir -p /etc/modules-load.d
+
+Create the configuration file:
+
+    sudo echo "zfs" | sudo tee /etc/modules-load.d/01-zfs.conf
+
+Reboot -- zfs.ko should be loaded automatically.
+
+Caution
+*******
+When the Clear Linux kernel is upgraded, DKMS will attempt to rebuild your
+zfs module for the new kernel. If you boot a new kernel and cannot find
+your zpools:
+
+- DKMS may not have rebuilt the module
+- DKMS may not have autoinstalled the module
+- The new kernel might introduce breaking changes that prevent zfs from compiling
+
+To fix this situation, you may have to recompile zfs.ko with the new kernel code.
+ZFS *might* not compile at all with the new kernel.
+
+**So, be sure you don't put anything on that ZFS pool that you would need
+in order to rebuild kernel modules.**
+
```

Comment:
Should add that you can check the results of dkms like this:
```
systemctl status dkms-new-kernel.service
```
or by checking the journal.

Reply:
Added